test/Dialect/FIRRTL/extract-instances.mlir

```diff
@@ -0,0 +1,342 @@
+    firrtl.strictconnect %bb_in, %invalid_ui8 : !firrtl.uint<8>
+    // CHECK: firrtl.connect %out, %bb_0_out
+    // CHECK: firrtl.connect %bb_0_in, %in
+    firrtl.connect %out, %bb_out : !firrtl.uint<8>, !firrtl.uint<8>
+    firrtl.connect %bb_in, %in : !firrtl.uint<8>, !firrtl.uint<8>
+  }
+  // CHECK-LABEL: firrtl.module private @DUTModule
+  // CHECK-SAME: out %bb_0_in: !firrtl.uint<8>
+  // CHECK-SAME: in %bb_0_out: !firrtl.uint<8>
+  firrtl.module private @DUTModule(in %in: !firrtl.uint<8>, out %out: !firrtl.uint<8>) attributes {annotations = [{class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+    // CHECK-NOT: firrtl.instance bb @MyBlackBox
+    // CHECK: %mod_in, %mod_out, %mod_bb_0_in, %mod_bb_0_out = firrtl.instance mod sym [[WRAPPER_SYM:@.+]] @BBWrapper
+    // CHECK-NEXT: firrtl.strictconnect %bb_0_in, %mod_bb_0_in
+    // CHECK-NEXT: firrtl.strictconnect %mod_bb_0_out, %bb_0_out
+    %mod_in, %mod_out = firrtl.instance mod @BBWrapper(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>)
+    firrtl.connect %out, %mod_out : !firrtl.uint<8>, !firrtl.uint<8>
+    firrtl.connect %mod_in, %in : !firrtl.uint<8>, !firrtl.uint<8>
+  }
+  // CHECK-LABEL: firrtl.module @ExtractBlackBoxesSimple
+  firrtl.module @ExtractBlackBoxesSimple(in %in: !firrtl.uint<8>, out %out: !firrtl.uint<8>) {
+    // CHECK: %dut_in, %dut_out, %dut_bb_0_in, %dut_bb_0_out = firrtl.instance dut sym {{@.+}} @DUTModule
+    // CHECK-NEXT: %bb_in, %bb_out = firrtl.instance bb sym [[BB_SYM:@.+]] @MyBlackBox
+    // CHECK-NEXT: firrtl.strictconnect %bb_in, %dut_bb_0_in
+    // CHECK-NEXT: firrtl.strictconnect %dut_bb_0_out, %bb_out
+    %dut_in, %dut_out = firrtl.instance dut @DUTModule(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>)
+    firrtl.connect %out, %dut_out : !firrtl.uint<8>, !firrtl.uint<8>
+    firrtl.connect %dut_in, %in : !firrtl.uint<8>, !firrtl.uint<8>
+  }
+  // CHECK: sv.verbatim "
+  // CHECK-SAME{LITERAL}: bb_0 -> {{0}}.{{1}}.{{2}}\0A
+  // CHECK-SAME: output_file = #hw.output_file<"BlackBoxes.txt", excludeFromFileList>
+  // CHECK-SAME: symbols = [
+  // CHECK-SAME: @DUTModule
+  // CHECK-SAME: #hw.innerNameRef<@DUTModule::[[WRAPPER_SYM]]>
+  // CHECK-SAME: #hw.innerNameRef<@ExtractBlackBoxesSimple::[[BB_SYM]]>
+  // CHECK-SAME: ]
+}
+
+//===----------------------------------------------------------------------===//
+// ExtractBlackBoxes Simple (modified)
+// ExtractBlackBoxes RenameTargets (modified)
+//===----------------------------------------------------------------------===//
+
+// CHECK: firrtl.circuit "ExtractBlackBoxesSimple2"
+firrtl.circuit "ExtractBlackBoxesSimple2" attributes {annotations = [{class = "firrtl.transforms.BlackBoxTargetDirAnno", targetDir = "BlackBoxes"}]} {
+  // CHECK: firrtl.nla @nla_1 [#hw.innerNameRef<@ExtractBlackBoxesSimple2::@bb>, @MyBlackBox]
+  // CHECK-NOT: firrtl.nla @nla_2
+  // CHECK-NOT: firrtl.nla @nla_3
+  firrtl.nla @nla_1 [#hw.innerNameRef<@BBWrapper::@bb>, @MyBlackBox]
```

Comment:
It seems like `@nla_1` should get remove, too (made local) due to the motion of `@MyBlackBox` into `@BBWrapper`?

Reply:
This one is special because it is explicitly targeting the `@MyBlackBox` module, not just the the instance. So this stays an NLA (because of the `@MyBlackBox` segment), but gets rerooted in `@ExtractBlackBoxesSimple2`. The corresponding annos were:

```json
// nla_1
{
  "class":"sifive.enterprise.firrtl.SimpleAnnotation",
  "target":"~TestHarness|BBWrapper/bb:MyBlackBox"
}
// nla_2
{
  "class":"sifive.enterprise.firrtl.SimpleAnnotation",
  "target":"~TestHarness|DUTModule/mod:BBWrapper>bb"
}
// nla_3
{
  "class":"sifive.enterprise.firrtl.SimpleAnnotation",
  "target":"~TestHarness|TestHarness/dut:DUTModule/mod:BBWrapper>bb"
}
```